drivers/adc/adc_stm32.c

```diff
@@ -633,8 +632,6 @@ static int start_read(const struct device *dev,
 		adc_stm32_enable(adc);
 		return -EINVAL;
 	}
-	/* re-enable ADC after changing the OVS */
```

Comment:
Shouldn't we remove the adc_stm32_enable in case of invalid oversampling as well (L632).
Sorry, I'm commenting code which you didn't touched, but while we're at..

Reply:
That's a good point. I kept it to always leave the `start_read()` function with the ADC enabled, just like it was before. But if it is fine changing this behavior, I can definitely remove it.